akka-remote/src/main/scala/akka/remote/artery/RemoteInstrument.scala

```diff
@@ -157,9 +157,9 @@ private[remote] final class RemoteInstruments(
     }
   }
 
-  def deserialize(inboundEnvelope: InboundEnvelope): Unit = {
+  def deserialize(inboundEnvelope: InboundEnvelope, positionOfMetaData: Int): Unit = {
```

Comment:
Might be easier to follow if more of the access to byte buffer was handled by the envelope rather than lots of areas of the code manipulating it

Reply:
I agree that more refactoring of those classes can be done to make it more clear, but leaving that for now.

I have added `byteBufferStartingAtMetaData` to the `InboundEnvelope` to avoid using the positionOfMetaData from the outside.